src/distributed_physical_optimizer_rule.rs

```diff
@@ -268,6 +275,13 @@ impl DistributedPhysicalOptimizerRule {
             Ok(Transformed::new(node, true, TreeNodeRecursion::Jump))
         })?;
 
+        // The head stage is executable, and upon execution, it will lazily assign worker URLs to
+        // all tasks. This must only be done once, so the executable StageExec must only be called
+        // once on 1 partition.
+        if depth == 0 && distributed.data.output_partitioning().partition_count() > 1 {
+            distributed.data = Arc::new(CoalescePartitionsExec::new(distributed.data));
+        }
```

Comment:
Question: What did we do before this PR? I am asking because I do not see any removed or modified code in this PR